src/utils/compareDates.js

```diff
@@ -0,0 +1,33 @@
+const compareDates = (date) => {
+    let dateSplit = date.split("-");
+    let month = parseInt(dateSplit[1]);
+    let day = parseInt(dateSplit[2]);
+
+    if ((month === 12 && day >= 22) || (month === 1 && day <= 19)) {
```

Comment:
I've still been thinking about this and trying to come up with a more concise way to do it, but I think this is the pretty concise given the number of details you need to consider. I will let you know if I have any ideas!